### PR TITLE
Fix/48736 fix settings header

### PR DESCRIPTION
**Pull request by ChavdaSachin**
### Fixed Issues

$ https://github.com/Expensify/App/issues/48736
PROPOSAL: https://github.com/Expensify/App/issues/48736#issuecomment-2335177712

### Tests

Same as QA Steps
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as QA Steps
### QA Steps

1. Open Settings
2. Scroll the Settings page.
3. Verify Header stays in place and scrollbar is hidden.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/d93fcf11-3bbc-4916-a69b-2639afdb0996

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/71d9a7b7-f285-4a34-a342-ae93e81e91f4

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/29841554-6ea9-4e45-9f65-2a682dee27ba

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/5a58cee1-d76e-49e0-bd58-b4f775e5101a

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/892c3ff6-184f-4eb5-a7c8-b605d0473e06

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/08bf986f-0c83-4ccc-8888-263e0d3cc2b4

</details>

### CODE REVIEW

**Comment by ChavdaSachin** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ChavdaSachin** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ChavdaSachin** [comment]
Tagging @Expensify/design for design review.

**Comment by ChavdaSachin** [comment]
recheck

**Comment by ikevin127** [comment]
> Tagging @Expensify/design for design review.

@ChavdaSachin I think we're good in terms of design as that was already determined in the issue. Regarding the CLA check failing, you're all good - that's expected. I'm working on the PR Reviewer Checklist as we speak 👍

**Review by ikevin127 — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/d87ab1e9-534e-41bc-bfe7-ecaecb28cf2c


</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/655cbdc6-81b4-432f-ad6a-234f2e5ec7f5


</details>

<details>
<summary>iOS: Native</summary>


https://github.com/user-attachments/assets/5428b7e1-4b0d-442a-8ab3-352d590606c1



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/c07ae92a-f797-49a6-bf58-b663cf380773



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/user-attachments/assets/6f42ddaa-aad7-4c9d-bb12-11629bb2b49d



</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/3fac89ae-b383-46ae-bfb0-073291884439




</details>

**Comment by dubielzyk-expensify** [comment]
This is mostly looking good, but the top header seems to be too big: 
CleanShot 2024-09-10 at 10 24 30@2x

I think it should follow the same height as all other page headers which is `80px`

There also seem to be a padded top instead of the whole page header being padded:
CleanShot 2024-09-10 at 10 22 38@2x
CleanShot 2024-09-10 at 10 25 09@2x


Other page header for comparison:
CleanShot 2024-09-10 at 10 23 15@2x


cc @Expensify/design for gutcheck here

**Comment by shawnborton** [comment]
I agree with those points Jon. It might be more difficult to get the height the same size here since the avatar we're using on settings is 52x52 instead of 40x40. That being said, we could probably just make the top area use a fixed height of 80px and vertically center everything? Or reduce the avatar down to 40x40. Thoughts?

**Comment by dubielzyk-expensify** [comment]
> That being said, we could probably just make the top area use a fixed height of 80px and vertically center everything? Or reduce the avatar down to 40x40. Thoughts?

I like that idea

**Comment by ChavdaSachin** [comment]
@dubielzyk-expensify check this screenshot with header height 80px with avatar size 52*52 and removed top padding from page wrapper. 
Does it look good to you?
Screenshot 2024-09-10 at 3 13 27 PM

Screenshot 2024-09-10 at 3 16 57 PM

**Comment by shawnborton** [comment]
I think that feels okay to me. Would love to see a quick video of jumping between Inbox, Search, and then Settings to see how that top part feels as you change pages.

**Comment by ChavdaSachin** [comment]
Here's the video you asked for.

https://github.com/user-attachments/assets/1b7a8c2f-96b4-40ce-bd7d-cdb313ccd96d

**Comment by ChavdaSachin** [comment]
Check this

Screenshot 2024-09-10 at 4 00 44 PM

https://github.com/user-attachments/assets/afc481bc-0a13-4741-bdf5-38b453c59145

**Comment by shawnborton** [comment]
Dang, it does feel kinda nice that the avatar stays in exactly the same spot across the app pages. Let's see what the rest of the @Expensify/design gang thinks though!

**Comment by ChavdaSachin** [comment]
If we decide to go with Avatar size 40*40, we additionally have to modify skeleton view for the same.
Heres what I came up with,
<details><summary>before changes</summary>
Screenshot 2024-09-10 at 4 34 25 PM
Screenshot 2024-09-10 at 4 34 36 PM

</details>

<details><summary>after changes</summary>
Screenshot 2024-09-10 at 4 29 12 PM
Screenshot 2024-09-10 at 4 33 28 PM

</details>

Code changes
- from
```js
<Rect
    x={startPositionX + avatarPlaceholderRadius + styles.gap3.gap}
    y="11"
    width="45%"
    height="8"
/>
<Rect
    x={startPositionX + avatarPlaceholderRadius + styles.gap3.gap}
    y="31"
    width="55%"
    height="8"
/>
```
- To
```js
<Rect
    x={startPositionX + avatarPlaceholderRadius + styles.gap3.gap}
    y="6"
    width="45%"
    height="8"
/>
<Rect
    x={startPositionX + avatarPlaceholderRadius + styles.gap3.gap}
    y="26"
    width="55%"
    height="8"
/>
```
and changed Avatar size passed for skeleton view, tell me what you think about it.

**Comment by dannymcclain** [comment]
>Dang, it does feel kinda nice that the avatar stays in exactly the same spot across the app pages.

Yeah I agree, this does actually feel pretty nice.

I think I was originally kind of against the 40x40 avatar there, but seeing it in this video looks pretty good. I think it feels a _little_ small next to the type lockup next to it, but I'm definitely down to give it a go!

That would once again simplify the amount of avatar sizes we're using in the app, and I agree that switching between tabs feels nice with the avatar always staying in the exact same spot.

---

🫣 I'm curious what @Expensify/design thinks of these `#tweaker` ideas...

| Before | After |
|--------|--------|
| CleanShot 2024-09-10 at 08 23 26@2x | CleanShot 2024-09-10 at 08 23 15@2x | 

Basically I'm:
- making the avatar `40x40`
- updating the name text to be `15px` instead of `17px`
- decreasing the gap between the name and the email address to be `0` instead of `4px`

This might be going too far, so just tell me to go away if that's the case 😂. Just trying to create more of a tight little lockup for the avatar/profile info in that top corner.


https://github.com/user-attachments/assets/c6c87756-0c29-413d-8d81-26134a6c0180

**Comment by shawnborton** [comment]
Hmm actually... should the name use our h2 styles to match other pages where we use New Kansas to say Inbox, Search, etc?! 

Otherwise I feel like I like 17px font size as that's our "page header" font size, and this feels like a page header to me. Super down with 40x40 though!

**Comment by dannymcclain** [comment]
Interesting Shawn! I kinda like that idea, but I do worry we might be straying from the path of this PR a little bit 😅

Here's a little mock showing that idea with the rest of the headers:

CleanShot 2024-09-10 at 08 46 57@2x

**Comment by shawnborton** [comment]
Hmm yeah, I think I like just the good ol' `Page Header` 17px the best, with a 40px avatar. That gives us a smaller amount of changes in this PR too as you mention (since we're startin' to cruise down some side streets)

**Comment by dannymcclain** [comment]
That sounds right to me.

**Comment by ChavdaSachin** [comment]
Changes are committed according to design team inputs.
Commit ready for review.
cc. @marcochavezf , @ikevin127

**Comment by ikevin127** [comment]
I will look into this once again to confirm everything works as expected 👍

**Comment by dubielzyk-expensify** [comment]
I vastly prefer the avatar at `40x40` also due to the fact that the vertical alignment feels better: 
CleanShot 2024-09-11 at 09 28 57@2x

**Comment by dubielzyk-expensify** [comment]
Checked it out locally and it looks great! Thanks for sweating the details with us here @ChavdaSachin 🥲

**Comment by ChavdaSachin** [comment]
Pleasure is all mine.
Despite the fact - working with design team shoulder to shoulder is a quite rare opportunity 😄.

**Comment by ChavdaSachin** [comment]
# idea

In order to keep app layout more consistent I propose to hide scrollbar from remaining 2 LHN pages as well
- LHN search page
- Workspace switcher
  
<details><summary>Screenshots</summary> 
Screenshot 2024-09-11 at 5 46 18 AM
Screenshot 2024-09-11 at 5 46 34 AM

</details>

if @expensify/design think this idea appropriate, feel free to raise an issue and assign me.

**Comment by ikevin127** [comment]
> In order to keep app layout more consistent I propose to hide scrollbar from remaining 2 LHN pages as well

I think we've already gone out of the scope enough with this PR, let's keep it contained for now and revisit if needed.

##

@marcochavezf 🟢 I just double checked the PR after the latest design adjustments and everything works and looks as expected - we're good to merge here from my side!

**Comment by dubielzyk-expensify** [comment]
Love coming to the table with ideas! 😄

I think for this one, there's quite a few considerations into accessibility as each platform does different things to the scroll bar. MacOS hides them by default, while some browsers override this etc. So I think we'll take this back and have a think, but not implement this for now cause we need to look at the ripple effects here. Again, love the idea and if we do it, we'll reach out 👍

**Comment by shawnborton** [comment]
Agree that we can discuss the scrolling issues as a separate issue/follow up. Glad we resolved the avatar size and fixed header area though, LGTM.

**Review by marcochavezf — APPROVED** [review]
LGTM, thanks!
